docs/en/operations/settings/settings.md

```diff
@@ -404,6 +404,38 @@ Possible values:
 
 Default value: 0.
 
+## partial_merge_join_optimizations {#partial_merge_join_optimizations}
+
+Enables of disables optimizations in partial merge join algorithm for [JOIN](../../sql-reference/statements/select/join.md) queries.
+
+Optimizations:
+
+- Skipping of  unneeded right blocks instead of pass through them.
+
+Possible values:
+
+-   0 — Optimizations disabled.
+-   1 — Optimizations enabled.
+
+Default value: 1.
+
+## partial_merge_join_rows_in_right_blocks {#partial_merge_join_rows_in_right_blocks}
+
+Limits a size of right-hand join data blocks.
```

Comment:
```suggestion
Limits size of right-hand join data blocks in partial merge join algorithm for [JOIN](../../sql-reference/statements/select/join.md) queries.
```

Reply:
It's possible to add that `you probably do not need to change default value`.

Reply:
@4ertus2 that's true for most settings, it would be too verbose to always point it out